### PR TITLE
[Form] Add missing return types to interfaces

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | https://github.com/symfony/symfony/issues/47551#issuecomment-1429726162, Continuation of #49383
| License       | MIT
| Doc PR        | N/A

I've picked a smaller slice this time, hoping it'll make it easier to get a green CI. 🤞🏻

### CODE REVIEW

File: src/Symfony/Component/Form/DataMapperInterface.php [hunk 1]
```diff
@@ -52,9 +54,11 @@ public function mapDataToForms(mixed $viewData, \Traversable $forms);
      * The model data can be an array or an object, so this second argument is always passed
      * by reference.
      *
-     * @param FormInterface[]|\Traversable $forms    A list of {@link FormInterface} instances
-     * @param mixed                        $viewData The compound form's view data that get mapped
-     *                                               its children model data
+     * @param \Traversable<mixed, FormInterface> $forms    A list of {@link FormInterface} instances
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/DataMapperInterface.php]: do we really have to use mixed for the key? can't we omit it?

File: src/Symfony/Component/Form/DataMapperInterface.php [hunk 2]
```diff
@@ -52,9 +54,11 @@ public function mapDataToForms(mixed $viewData, \Traversable $forms);
...
+     * @param mixed                              $viewData The compound form's view data that get mapped
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/DataMapperInterface.php]: missing`&` I suppose?

File: src/Symfony/Component/Form/DataMapperInterface.php (hunk 1 above)
Comment (derrabus) [reply src/Symfony/Component/Form/DataMapperInterface.php]: All static analyzers use stubs with two template variables afaik. And `mixed` is the closest thing to "I don't care" I could come up with. 🤷

**Comment by nicolas-grekas** [comment]
Thank you @derrabus.

File: src/Symfony/Component/Form/DataMapperInterface.php (hunk 1 above)
Comment (derrabus) [reply src/Symfony/Component/Form/DataMapperInterface.php]: Omitting the key template variable seems to work though:
* https://phpstan.org/r/ed0300e9-008a-4bfc-84b4-3d54c7be8bfb
* https://psalm.dev/r/524d8b2350

TIL 🤓

**Pull request merged**